Added new docs contribution page (#6080)
mattermost/docs: source/guides/contribute-to-documentation.rst

Contribute to this site
========================

Mattermost has a diverse community that extends well beyond those who write code for our core projects. We value every contributor who helps make Mattermost better for everyone, from the community members who translate the Mattermost platform into languages beyond English, to those who write documentation and beyond. 

> If you’re interested in contributing to Mattermost, want to help improve the product documentation?

Reviewer: ```suggestion
If you’re interested in contributing to Mattermost, why not help improve the product documentation?
```

How to get started
------------------

> The fastest way to get started with a documentation contribution is to find something you want to change in the documentation, such as a typo or link fix, or updating content for clarity. Select **Edit** in the top-right corner of the page. 

Reviewer 2: ```suggestion
The fastest way to get started with a documentation contribution is to find something you want to change in the documentation, such as a typo or link fix, or updating content for clarity. Once you've found a page you want to update, select **Edit** in the top-right corner of that page. 
```

> The fastest way to get started with a documentation contribution is to find something you want to change in the documentation, such as a typo or link fix, or updating content for clarity. Select the **Edit** button in the top right corner of the page. 

Reviewer: ```suggestion
The fastest way to get started with a documentation contribution is to find something you want to change in the documentation, such as a typo or link fix, or updating content for clarity. Select **Edit** in the top-right corner of the page. 
```

.. raw:: html

   <div style="position: relative; padding-bottom: 50%; height: 0; overflow: hidden; max-width: 100%; height: auto;">
      <iframe src="https://www.youtube.com/embed/V8ROl9wiHr8" alt="Video on creating pull requests in GitHub" frameborder="0" allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 95%;"></iframe>
   </div>

See our `contribution <https://developers.mattermost.com/contribute/why-contribute/#you-want-to-help-with-content>`__ documentation to learn more about the contribution opportunities available.

Where to find more information
------------------------------

If you'd like to take on a project with larger scope across multiple pages at a time, you'll want to be able to generate the documentation locally on demand to review your changes. See the product documentation `README <https://github.com/mattermost/docs#readme>`__ documentation to learn how to build the product documentation locally on your own system. 

> Contributing to Mattermost developer or API reference documentation? See the additional README links below for details on generating documentation locally:

Reviewer 2: ```suggestion
Are you interested in contributing to the Mattermost developer or API reference documentation? See the additional README links below for details on generating documentation locally:
```
